### ISSUE TITLE
Resolution of test targets

### ISSUE THREAD

**Issue by ex0ns** [com-lihaoyi/mill#2602]
I think this is related to 0.11.0 as well, with a simple `build.sc`:

```scala
import mill._
import scalalib._


object core extends ScalaModule {
  def scalaVersion = "2.13.11"

  object test extends ScalaTests with TestModule.ScalaTest {

  }
}
```

The resolution of the tests targets is as follow:

```
mill resolve __.test
[1/1] resolve 
core.test
core.test.test
```

This is an issue when running the tests (mostly in parallel ? I can't confirm for sequential), for instance using `mill -j 0 __.test` because it runs all the tests twice and can has some unexpected race condition (when using zio-test for instance, the test often crashes because of a missing report file), furthermore the output of the test command is (without any surprise) duplicated and hence hard to follow/read as multiple execution of the same tests are happening at the same time.

A workaround has been to change my build process to call `mill -j 0 __.test.test` instead as it only resolve the inner target.
I think that we should take care of de-duplicating targets because running the build, this could improve the performances and reliability of mill.

For reference here is the output of the inspect command for this build:

```
core.test.test(build.sc:9)
    Discovers and runs the module's tests in a subprocess, reporting the
    results to the console.
    @see [[testCached]]

    args <str>...

Inputs:
    core.test.testUseArgsFile
    core.test.forkArgs
    core.test.forkArgs
    core.test.testFramework
    core.test.runClasspath
    core.test.compile
    core.test.runClasspath
    core.test.forkEnv
    core.test.forkWorkingDir

core.test.test(build.sc:9)
    Discovers and runs the module's tests in a subprocess, reporting the
    results to the console.
    @see [[testCached]]

    args <str>...

Inputs:
    core.test.testUseArgsFile
    core.test.forkArgs
    core.test.forkArgs
    core.test.testFramework
    core.test.runClasspath
    core.test.compile
    core.test.runClasspath
    core.test.forkEnv
    core.test.forkWorkingDir
```

Here we can see that both target are indeed referencing the same build (same line, same inputs, same doc).

**Comment by lefou** [comment]
So the with duplicated tests is with Mill 0.10.12 or 0.11.0?

_Can you please ask the second topic as a question in the discussions? Otherwise we loose track of the original issue and nobody will find the potential interesting answer to the other question._

**Comment by ex0ns** [comment]
I didn't test it with 0.10.12, maybe it's doing the same, I don't know, I noticed that when upgrading project to mill 0.11.0

Good point for the second question, will update

**Comment by lefou** [comment]
Version `0.11.0-11-e5a399` should fix this issue.

**Comment by ex0ns** [comment]
Hi, thanks for the quick fix, this seems to work correctly, even when using the `-j 0` flag I do not get duplicated output. 

I can also confirm that `mill inspect __.test` do not show the tests twice, which is great.

I don't know if this is wanted or not, but the `resolve` command still shows them all,  this might not be a bug at all, as I think that those tasks actually exist, even if they are duplicated:

```
ill resolve __.test
[1/1] resolve 
SSLExpChecker.backend.test
SSLExpChecker.backend.test.test
SSLExpChecker.database.test
SSLExpChecker.database.test.test
SSLExpChecker.frontend.test
SSLExpChecker.frontend.test.test
```
